src/ReactiveDomain.Testing/Messaging/ConcurrentMessageQueue.cs

```diff
@@ -57,7 +57,10 @@ public ConcurrentMessageQueue(string name, IEnumerable<T> items)
         public void AssertEmpty()
         {
             if (!IsEmpty)
-                throw new Exception($" {_name} Queue not Empty.");
+            {
+                var msg = this.DequeueNext<T>();
```

Comment:
This is a side effect since it modifies the queue. We should use the first message on the queue without dequeueing it when constructing the message.

Reply:
Hmmm...  I am throwing the same exception so the test will not pass.  I just changed the text inside.
I did not see another way to read the next message nor did I find a 'Requeue'.

Reply:
You can't count on this being used a particular way to make a side effect be OK. You'll need to use the first message on the collection without removing it from the queue.